cowprotocol/cowswap
# fix: :bug: update getAppDataHash to rely on appDataInfoAtom

# Summary

Fixes the issue regarding quote endpoints incorrectly identifying themselves, causing security systems to assume we are being hit by a denial of service attack.

This change does two things;
* When `useAppData` runs, update a local variable to set the app data hash
* In `getAppDataHash` use this value if present

Why? Because not everywhere we use this value do we have the react available (i.e. api fetchers). Since we are not using jotai v2+, we cannot use the store api to extract this value properly. As such, this solution should serve us well.

# To Test

1. Open CoW Swap
2. Enter currency values to get a quote endpoint call made
3. Go to console, search for useAppData
4. appDataHash you see there, copy it somewhere
5. Go to network tab
6. Find a recent quote endpoint call
7. Go to request tab to see request body
8. appData field you see here should be identical to appDataHash you have saved

# Background
https://cowservices.slack.com/archives/C0361CDG8GP/p1682539030086899
